sigstore/_verify.py

```diff
@@ -155,6 +157,17 @@ def verify(
                 reason=f"Subject name does not contain identity: {cert_email}"
             )
 
+    if cert_oidc_issuer is not None:
+        issuers = [
+            issuer.value
+            for issuer in cert.issuer.get_attributes_for_oid(OID_ORGANIZATION_NAME)
```

Comment:
Yeah, it should be the organization name. I guess that also makes the flag somewhat confusing, since the "issuer" in the OIDC context is the org name in the X.509 context...

Reply:
Interestingly, the help text for `cosign` says this:

> the OIDC issuer expected in a valid Fulcio certificate, e.g. https://token.actions.githubusercontent.com or https://oauth2.sigstore.dev/auth

Which doesn't line up with the organization name (`sigstore.dev`). I wonder if that help text is wrong or I've done this incorrectly.

Reply:
I can't really tell what `cosign` is doing. I tried typing random args for `--cert-oidc-issuer` and it didn't seem to care. Will have to look at the code tomorrow and figure out what it's doing.

Reply:
Oh, huh -- yeah, I'd expect it to be the OIDC issuer, not just the bare `sigstore.dev` domain.

Reply:
Ditto...

Reply:
Yeah, our expected issuer doesn't show up anywhere in the cert:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            3d:b8:37:67:e1:71:9a:41:ad:a3:a0:aa:8c:49:17:77:56:37:66:b7
        Signature Algorithm: ecdsa-with-SHA384
        Issuer: O = sigstore.dev, CN = sigstore-intermediate
        Validity
            Not Before: Jun  3 17:11:31 2022 GMT
            Not After : Jun  3 17:21:31 2022 GMT
        Subject: 
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:fb:0d:2d:65:07:ce:d2:d2:a8:72:9d:70:61:89:
                    ad:f3:34:3d:8c:21:5f:5c:42:86:dd:c0:97:8a:6f:
                    d3:73:86:c5:6f:c7:2e:8f:27:8e:2b:36:85:d3:cd:
                    ee:26:de:cc:42:81:d1:38:94:c5:8c:f9:95:9a:a5:
                    bd:8f:62:9a:95:0e:92:ae:fb:12:56:ea:39:0d:0e:
                    2e:9d:b5:a2:4b:fe:b8:46:49:a4:d8:d5:70:f4:78:
                    19:cb:ff:2a:88:56:32
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature
            X509v3 Extended Key Usage: 
                Code Signing
            X509v3 Subject Key Identifier: 
                1F:C1:DF:F9:0C:7A:1A:C9:AC:76:F5:7F:AB:46:F4:E8:C6:98:0B:AC
            X509v3 Authority Key Identifier: 
                DF:D3:E9:CF:56:24:11:96:F9:A8:D8:E9:28:55:A2:C6:2E:18:64:3F
            X509v3 Subject Alternative Name: critical
                email:william@yossarian.net
            1.3.6.1.4.1.57264.1.1: 
                github-sigstore-production
```

...so maybe this just doesn't work at all, at the moment?

Reply:
That was with `sigstore sign README.md --output-certificate`, which used the OAuth2/OIDC flow on my local machine. Possibly a production regression?

Reply:
For context: I did the OAuth2 flow using my GitHub account, not Google. I'll try Google now.

Reply:
This might be a quirk with GitHub logins that I wasn't aware of. I'll need to look into this, because `github-sigstore-production` is a client ID, not an issuer.

Reply:
Similar results with the Google provider:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            69:ac:04:c6:78:15:3f:05:a1:06:68:bc:f6:ae:95:4e:a2:2d:1d:6a
        Signature Algorithm: ecdsa-with-SHA384
        Issuer: O = sigstore.dev, CN = sigstore-intermediate
        Validity
            Not Before: Jun  3 17:57:34 2022 GMT
            Not After : Jun  3 18:07:34 2022 GMT
        Subject: 
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:76:f5:18:58:84:25:71:34:90:38:ae:cb:c2:ae:
                    11:67:dd:e9:8e:58:44:3e:27:f8:37:1d:59:9a:95:
                    05:96:7f:65:7a:68:92:bc:99:a9:9c:66:99:86:30:
                    74:83:2b:86:54:db:4c:f6:ec:fc:7a:8a:46:17:a8:
                    98:fd:17:b1:51:87:58:02:33:e1:3f:80:52:6b:b7:
                    2c:87:74:da:6c:e0:51:94:24:1e:53:b9:2d:98:40:
                    bb:33:be:e8:9c:ce:36
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature
            X509v3 Extended Key Usage: 
                Code Signing
            X509v3 Subject Key Identifier: 
                76:DF:69:A7:40:6F:51:F0:26:8C:16:D4:B8:E6:C0:B2:A9:A5:54:96
            X509v3 Authority Key Identifier: 
                DF:D3:E9:CF:56:24:11:96:F9:A8:D8:E9:28:55:A2:C6:2E:18:64:3F
            X509v3 Subject Alternative Name: critical
                email:william.woodruff@trailofbits.com
            1.3.6.1.4.1.57264.1.1: 
                google-sigstore-production
```

Reply:
Thanks, investigating now

Reply:
this is a regression, huge thanks for finding this. it's being fixed now

Reply:
No problem! Glad we were able to catch it!

Reply:
can you try again now? it should be fixed

Reply:
That definitely looks better:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            0a:f1:ab:11:01:7e:9e:34:1d:38:16:5c:4d:1d:ee:ff:52:32:48:4f
        Signature Algorithm: ecdsa-with-SHA384
        Issuer: O = sigstore.dev, CN = sigstore-intermediate
        Validity
            Not Before: Jun  3 18:59:46 2022 GMT
            Not After : Jun  3 19:09:46 2022 GMT
        Subject: 
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:4b:24:33:be:80:22:18:53:80:af:92:46:a9:da:
                    ae:e5:90:91:fe:20:6b:5a:a1:e4:04:e7:53:cc:1c:
                    d9:7e:b6:5b:dd:2e:a9:cd:f3:4c:ba:4d:fa:c6:7b:
                    da:2c:ee:f0:f2:66:d7:fd:7d:10:01:78:f7:b1:70:
                    81:49:40:ed:c8:6c:28:78:a8:30:32:d4:0b:74:65:
                    44:d6:de:5b:ce:2e:12:26:81:26:b7:0c:29:5e:b6:
                    f8:f1:7a:7a:11:ef:2e
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature
            X509v3 Extended Key Usage: 
                Code Signing
            X509v3 Subject Key Identifier: 
                61:C1:09:A5:71:6C:A9:ED:A1:00:05:FF:A0:8E:42:82:0F:6D:B2:D7
            X509v3 Authority Key Identifier: 
                DF:D3:E9:CF:56:24:11:96:F9:A8:D8:E9:28:55:A2:C6:2E:18:64:3F
            X509v3 Subject Alternative Name: critical
                email:william@yossarian.net
            1.3.6.1.4.1.57264.1.1: 
                https://github.com/login/oauth
```

So I see `https://github.com/login/oauth` now, which I _think_ is correct?

Reply:
Yep, that looks correct to me.

Reply:
Just to clarify: we're checking for the organization name in the current code, but it looks like the OIDC issuer URL is only present in a custom X.509 v3 extension. Should it also be present in the X.509 issuer (as `ON` or `OU` perhaps?)

Reply:
That'd probably be overloading the x.509 issuer. That is supposed to only refer to the issuing certificate authority. for code signing certificates, there's not really another great place to put extra information besides in custom OIDs.

Reply:
Gotcha, makes sense. Just wanted to make sure we were looking in the definitive location!